iscript/src/iscript/hardened_sign.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python
+"""Functions to sign packages with hardened runtime"""
+
+import asyncio
+import logging
+import os
+import shutil
+from glob import glob
+from pathlib import Path
+
+from iscript.autograph import sign_langpacks
+from iscript.exceptions import IScriptError
+from iscript.mac import (
+    create_pkg_files,
+    copy_pkgs_to_artifact_dir,
+    download_requirements_plist_file,
+    extract_all_apps,
+    filter_apps,
+    get_app_paths,
+    set_app_path_and_name,
+    sign_omnija_with_autograph,
+    sign_widevine_dir,
+    tar_apps,
+    unlock_keychain,
+    update_keychain_search_path,
+)
+from iscript.util import get_sign_config
+from scriptworker_client.aio import download_file, raise_future_exceptions, retry_async
+from scriptworker_client.exceptions import DownloadError
+from scriptworker_client.utils import run_command
+
+log = logging.getLogger(__name__)
+
+
+async def download_signing_resources(hardened_sign_config, folder):
```

Comment:
Did you consider adding the full entitlements to the payload (as opposed to downloading it at runtime?). The former has a couple of benefits I can see:
* No network dependence (more reliable)
* More reproducible (no chance of the remote URL changing over time, eg: if it refers to a branch or tag name

Reply:
I think I'd still prefer having the exact file from the repo show up in the app entitlements without parsing and composing a new file. That way we "future proof" if the file ever changes to a different format (never doubt apple would do something like this lol)

Reply:
Just to be clear: I'm not suggesting that we need to reconstitute the file, but rather we put its entire contents into one field of the payload (which can then be written out if needed, or simply passed along to whatever function requires it as a string).

I'll leave this your call - but if we start having issues with intermittents with this we should consider it.